### DISCUSSION TITLE
Move up the Data Tree

### DISCUSSION THREAD

**Discussion by ghost** [rrousselGit/freezed#669] [Q&A]
I have the following scenario:

I have exercises that contain tasks. Now when I am on the exercise page it's no problem to navigate to the task page as I got access to my task right there. But if I want to reference the exercise from within the task page it becomes cumbersome. I know some libraries (not flutter libraries) that allow you to just call the .parent attribute on any given data model to get its parent.

Assume this model:

```
@freezed
class ExerciseModel with _$ExerciseModel {
  const factory ExerciseModel({
    required String id,
    required String name,
    required List<TaskModel> tasks,
  }) = _ExerciseModel;

  factory ExerciseModel.fromJson(Map<String, Object?> json) =>
      _$ExerciseModelFromJson(json);
}

@freezed
class TaskModel with _$TaskModel {
  const factory TaskModel({
    required String id,
    String? groupType,
    dynamic content,
  }) = _TaskModel;

  factory TaskModel.fromJson(Map<String, Object?> json) =>
      _$TaskModelFromJson(json);
}
```

So basically I can do this on the exercise page:

```
ExerciseModel exercise = ExerciseModel.fromJson(someJSON);
Navigator.of(context).push(TaskPage(task: exercise.tasks[0]));
```

But I want something like this as well on the task page:

```
@override
Widget build(BuildContext context) {
  ExerciseModel exercise = task.parent;

  return Text(exercise.name);
}
```

Is there something equivalent in freezed aswell? Or what would be the most elegant way of achieving a similar behavior in freezed for flutter?

**Comment by rrousselGit** [comment]
You can make your Freezed classes receive a "parent" value, and pass that value yourself.

**Reply by ghost** [reply]
1. I feel that would get out of hand really fast, especially when there is more than one level of nesting
2. Also would that approach at all be able to be automated? Or would I have to initiate the model, and then go and reference that instance in all sub models? Because that sounds messy.
